airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/utils/json_schema_helper.py

```diff
@@ -45,20 +45,27 @@ def get_property(self, path: List[str]):
             node = node["properties"][segment]
         return node
 
-    def get_type_for_key_path(self, path: List[str]):
+    def get_type_for_key_path(self, path: List[str]) -> Set[str]:
+        type_ = []
         try:
-            return self.get_property(path)["type"]
+            field = self.get_property(path)
+            type_ = field.get("format", field["type"])
+            if not isinstance(type_, List):
+                type_ = [type_]
         except KeyError:
-            return None
+            pass
+        return set(type_)
 
     def get_cursor_value(self, record, cursor_path):
         type_ = self.get_type_for_key_path(path=cursor_path)
         value = reduce(lambda data, key: data[key], cursor_path, record)
         return self.parse_value(value, type_)
 
     @staticmethod
-    def parse_value(value, type_):
-        if type_ in ("datetime", "date-time"):
+    def parse_value(value: Any, type_: Set[str]):
+        if type_ & {"datetime", "date-time"}:
```

Comment:
python is bonkers

Reply:
this can be expressed like this:
```python
if type_.intersect({"datetime", "date-time"}):
```

Reply:
I think I slightly prefer that just because it's less cognitive load (`intersect` tells you exactly what it is doing)